--- openpetra.freeze/csharp/ICT/Petra/Shared/Constants.cs	2014-07-24 14:18:45.000000000 +0200
+++ openpetra.patched/csharp/ICT/Petra/Shared/Constants.cs	2014-07-24 14:19:19.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       christiank
 //
-// Copyright 2004-2010 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -173,9 +173,6 @@
         public const String SYSDEFAULT_GLREFMANDATORY = "GLRefMandatory";
 
         /// <summary>ClientTask TaskGroups</summary>
-        public const String CLIENTTASKGROUP_DISCONNECT = "DISCONNECT";
-
-        /// <summary>ClientTask TaskGroups</summary>
         public const String CLIENTTASKGROUP_USERMESSAGE = "USERMESSAGE";
 
         /// <summary>ClientTask TaskGroups</summary>
