--- trunk/csharp/ICT/Common/Remoting/Shared/ConstantsAndTypes.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Common/Remoting/Shared/ConstantsAndTypes.cs	2014-05-10 06:53:13.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -56,6 +56,9 @@
 
         /// <summary>Remoting URL Identifiers</summary>
         public const String REMOTINGURL_IDENTIFIER_POLLCLIENTTASKS = "PollClientTasks";
+
+        /// <summary>ClientTask TaskGroups</summary>
+        public const String CLIENTTASKGROUP_DISCONNECT = "DISCONNECT";
     }
 
     /// an interface for cacheable tables manager
