--- before/csharp/ICT/Common/Logging.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Common/Logging.cs	2014-05-15 21:56:00.000000000 +0200
@@ -2,9 +2,9 @@
 // DO NOT REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 //
 // @Authors:
-//       christiank
+//       christiank, timop
 //
-// Copyright 2004-2011 by OM International
+// Copyright 2004-2014 by OM International
 //
 // This file is part of OpenPetra.org.
 //
