--- before/csharp/ICT/Petra/Server/app/Core/ServerManager.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Server/app/Core/ServerManager.cs	2014-05-15 21:56:00.000000000 +0200
@@ -68,6 +68,18 @@
         private IUserManager FUserManager;
 
         /// <summary>
+        /// get a casted version of the static variable
+        /// </summary>
+        public static TServerManager TheCastedServerManager
+        {
+            get
+            {
+                return (TServerManager)TheServerManager;
+            }
+        }
+
+
+        /// <summary>
         /// Initialises Logging and parses Server settings from different sources.
         /// </summary>
         public TServerManager() : base()
@@ -75,10 +87,6 @@
             // Create SystemDefaults Cache
             FSystemDefaultsCache = new TSystemDefaultsCache();
 
-            TRemoteLoader.CLIENTDOMAIN_DLLNAME = "Ict.Petra.Server.app.Core";
-            TRemoteLoader.CLIENTDOMAIN_CLASSNAME = "Ict.Petra.Server.App.Core.TClientDomainManager";
-            TClientAppDomainConnectionBase.ClientAppDomainConnectionType = typeof(TClientAppDomainConnection);
-
             TCacheableTablesManager.InitializeUnit();
             TCacheableTablesManager.GCacheableTablesManager = new TCacheableTablesManager(new TDelegateSendClientTask(TClientManager.QueueClientTask));
 
@@ -98,11 +106,9 @@
                 null);
 
             TClientManager.InitializeStaticVariables(FSystemDefaultsCache,
-                TCacheableTablesManager.GCacheableTablesManager,
                 FUserManager,
                 new TErrorLog(),
-                new TMaintenanceLogonMessage(),
-                new TClientAppDomainConnection());
+                new TMaintenanceLogonMessage());
 
             TTimedProcessing.DailyStartTime24Hrs = TAppSettingsManager.GetValue("Server.Processing.DailyStartTime24Hrs", "00:30");
 
