--- before/csharp/ICT/Petra/Server/lib/MFinance/AP/AP.EditSupplier.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Server/lib/MFinance/AP/AP.EditSupplier.cs	2014-05-15 21:56:00.000000000 +0200
@@ -59,7 +59,7 @@
     ///          However, Server Objects that derive from these objects and that
     ///          are also UIConnectors are feasible.
     ///</summary>
-    public class TSupplierEditUIConnector : TConfigurableMBRObject, IAPUIConnectorsSupplierEdit
+    public class TSupplierEditUIConnector : IAPUIConnectorsSupplierEdit
     {
         /// <summary>
         /// constructor
