--- before/csharp/ICT/Petra/Server/lib/MPartner/Common/ServerLookups.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Server/lib/MPartner/Common/ServerLookups.cs	2014-05-15 21:56:00.000000000 +0200
@@ -167,7 +167,7 @@
         /// access this partner</param>
         /// <returns>true if Partner was found in DB or Partner key = 0, otherwise false</returns>
         [RequireModulePermission("PTNRUSER")]
-        public static Boolean VerifyPartner(Int64 APartnerKey,
+        public static Boolean VerifyPartnerAndGetDetails(Int64 APartnerKey,
             out String APartnerShortName,
             out TPartnerClass APartnerClass,
             out Boolean AIsMergedPartner,
