--- before/csharp/ICT/Petra/Server/lib/MPartner/web/MergePartners.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Server/lib/MPartner/web/MergePartners.cs	2014-05-15 21:56:00.000000000 +0200
@@ -30,6 +30,7 @@
 using Ict.Common;
 using Ict.Common.DB;
 using Ict.Common.Verification;
+using Ict.Common.Remoting.Server;
 using Ict.Petra.Server.App.Core;
 using Ict.Petra.Server.App.Core.Security;
 using Ict.Petra.Server.MConference.Data.Access;
