--- before/csharp/ICT/Testing/lib/MFinance/server/Budget/Test.Budget.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Testing/lib/MFinance/server/Budget/Test.Budget.cs	2014-05-15 21:56:00.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2013 by OM International
+// Copyright 2004-2014 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -55,7 +55,6 @@
     public class TestBudget
     {
         private const int FLedgerNumber = 43;
-        private BudgetTDS FMainDS;
 
         /// <summary>
         /// TestFixtureSetUp
@@ -213,21 +212,31 @@
         /// <summary>
         /// test the budget autogeneration
         /// </summary>
-        [Test]
-        public void T1_AutoGenerationLoadData()
+        private BudgetTDS LoadData()
         {
-            FMainDS = new BudgetTDS();
+            BudgetTDS MainDS = new BudgetTDS();
 
-            FMainDS.Merge(TBudgetAutoGenerateWebConnector.LoadBudgetForAutoGenerate(FLedgerNumber));
+            MainDS.Merge(TBudgetAutoGenerateWebConnector.LoadBudgetForAutoGenerate(FLedgerNumber));
 
             //Not needed
-            FMainDS.RemoveTable("AGeneralLedgerMasterPeriod");
+            MainDS.RemoveTable("AGeneralLedgerMasterPeriod");
+
+            return MainDS;
+        }
+
+        /// <summary>
+        /// test the budget autogeneration
+        /// </summary>
+        [Test]
+        public void T1_AutoGenerationLoadData()
+        {
+            BudgetTDS MainDS = LoadData();
 
             string emptyTables = string.Empty;
 
-            foreach (DataTable tb in FMainDS.Tables)
+            foreach (DataTable tb in MainDS.Tables)
             {
-                if (FMainDS.Tables[tb.TableName].Rows.Count == 0)
+                if (MainDS.Tables[tb.TableName].Rows.Count == 0)
                 {
                     emptyTables += tb.TableName + "; ";
                 }
@@ -242,11 +251,12 @@
         [Test]
         public void T2_AutoGenerationGenBudget()
         {
-            //FMainDS loaded in previous test
-            int budgetSequence = FMainDS.ABudget.Count > 0 ? FMainDS.ABudget[0].BudgetSequence : 0;
+            BudgetTDS MainDS = LoadData();
+
+            int budgetSequence = MainDS.ABudget.Count > 0 ? MainDS.ABudget[0].BudgetSequence : 0;
             string forecastType = MFinanceConstants.FORECAST_TYPE_BUDGET;
 
-            if (FMainDS.ABudget.Count == 0)
+            if (MainDS.ABudget.Count == 0)
             {
                 return;
             }
@@ -294,15 +304,16 @@
         [Test]
         public void T4_AutoGenerationSetBudgetAmount()
         {
-            //FMainDS loaded in previous test
-            if (FMainDS.ABudget.Count == 0)
+            BudgetTDS MainDS = LoadData();
+
+            if (MainDS.ABudget.Count == 0)
             {
                 return;
             }
 
-            int budgetSequence = FMainDS.ABudget[0].BudgetSequence;
+            int budgetSequence = MainDS.ABudget[0].BudgetSequence;
 
-            ABudgetPeriodRow bPRow = (ABudgetPeriodRow)FMainDS.ABudgetPeriod.Rows.Find(new object[] { budgetSequence, 1 });
+            ABudgetPeriodRow bPRow = (ABudgetPeriodRow)MainDS.ABudgetPeriod.Rows.Find(new object[] { budgetSequence, 1 });
 
             if (bPRow == null)
             {
