--- trunk/csharp/ICT/Petra/Server/lib/MFinance/Gift/Gift.Transactions.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Server/lib/MFinance/Gift/Gift.Transactions.cs	2014-05-10 06:53:13.000000000 +0200
@@ -35,6 +35,7 @@
 using Ict.Common.Data;
 using Ict.Common.DB;
 using Ict.Common.Verification;
+using Ict.Common.Remoting.Server;
 using Ict.Petra.Server.App.Core;
 using Ict.Petra.Server.App.Core.Security;
 using Ict.Petra.Server.MFinance.Account.Data.Access;
