--- before/csharp/ICT/Testing/lib/MPartner/server/PartnerEdit/TestPartnerEdit.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Testing/lib/MPartner/server/PartnerEdit/TestPartnerEdit.cs	2014-05-15 21:56:00.000000000 +0200
@@ -63,20 +63,13 @@
     public class TPartnerEditTest
     {
         /// <summary>
-        /// use automatic property to avoid compiler warning about unused variable FServerManager
-        /// </summary>
-        private TServerManager FServerManager {
-            get; set;
-        }
-
-        /// <summary>
         /// open database connection or prepare other things for this test
         /// </summary>
         [TestFixtureSetUp]
         public void Init()
         {
             new TLogging("../../log/TestServer.log");
-            FServerManager = TPetraServerConnector.Connect("../../etc/TestServer.config");
+            TPetraServerConnector.Connect("../../etc/TestServer.config");
         }
 
         /// <summary>
@@ -727,6 +720,7 @@
             TSubmitChangesResult result;
             Int64 PartnerKey;
 
+            DBAccess.GDBAccessObj.BeginTransaction(IsolationLevel.Serializable);
             TPartnerEditUIConnector connector = new TPartnerEditUIConnector();
 
             PartnerEditTDS MainDS = new PartnerEditTDS();
@@ -776,6 +770,8 @@
 
             // check that Bank record is really deleted
             Assert.IsTrue(!TPartnerServerLookups.VerifyPartner(PartnerKey));
+
+            DBAccess.GDBAccessObj.CommitTransaction();
         }
 
         /// <summary>
