--- trunk/csharp/ICT/Common/Gettext.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Common/Gettext.cs	2014-05-10 06:53:13.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2014 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -65,11 +65,15 @@
             Thread.CurrentThread.CurrentUICulture = new CultureInfo(ALanguageCode);
 
             string ResourceDllFname = TAppSettingsManager.ApplicationDirectory +
-                                      "\\" + Thread.CurrentThread.CurrentUICulture.IetfLanguageTag + "\\OpenPetra.resources.dll";
+                                      Path.DirectorySeparatorChar +
+                                      Thread.CurrentThread.CurrentUICulture.Name +
+                                      Path.DirectorySeparatorChar +
+                                      "OpenPetra.resources.dll";
 
             if (File.Exists(ResourceDllFname))
             {
-                catalog = new GettextResourceManager("OpenPetra");
+                catalog = new GettextResourceManager("OpenPetra",
+                    TAppSettingsManager.ApplicationDirectory);
             }
             else
             {
