--- trunk/csharp/ICT/BuildTools/GenerateGlue/CollectConnectors.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/BuildTools/GenerateGlue/CollectConnectors.cs	2014-05-10 06:53:13.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -154,7 +154,7 @@
 //        TLogging.Log("implementing " + ANamespace);
         foreach (string key in AConnectors.Keys)
         {
-            if (key.StartsWith(ANamespace))
+            if (key.StartsWith(ANamespace) && (key.LastIndexOf(".") == ANamespace.Length))
             {
 //                TLogging.Log("   " + AConnectors[key]);
                 Result.Add(AConnectors[key]);
@@ -212,6 +212,11 @@
                 CSFiles = CSParser.GetCSFilesForDirectory(CSParser.ICTPath + "/Petra/Server/lib/M" + AModuleName,
                     SearchOption.AllDirectories);
             }
+            else if (AModuleName == "ServerAdmin")
+            {
+                CSFiles = CSParser.GetCSFilesForDirectory(CSParser.ICTPath + "/Petra/Server/app/Core",
+                    SearchOption.AllDirectories);
+            }
             else
             {
                 CSFiles = new List <CSParser>();
