--- before/csharp/ICT/Testing/lib/Common/DB/test.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Testing/lib/Common/DB/test.cs	2014-05-15 21:56:00.000000000 +0200
@@ -76,7 +76,7 @@
         [SetUp]
         public void Init()
         {
-            new TLogging("test.log");
+            new TLogging("../../log/test.log");
             new TAppSettingsManager("../../etc/TestServer.config");
 
             EstablishDBConnection();
