--- before/csharp/ICT/Petra/Shared/Constants.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Shared/Constants.cs	2014-05-15 21:56:00.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       christiank
 //
-// Copyright 2004-2010 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -172,9 +172,6 @@
         public const String SYSDEFAULT_GLREFMANDATORY = "GLRefMandatory";
 
         /// <summary>ClientTask TaskGroups</summary>
-        public const String CLIENTTASKGROUP_DISCONNECT = "DISCONNECT";
-
-        /// <summary>ClientTask TaskGroups</summary>
         public const String CLIENTTASKGROUP_USERMESSAGE = "USERMESSAGE";
 
         /// <summary>ClientTask TaskGroups</summary>
