--- before/csharp/ICT/Petra/Client/app/Core/ServerLookups.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Client/app/Core/ServerLookups.cs	2014-05-15 21:56:00.000000000 +0200
@@ -204,7 +204,7 @@
                 out Boolean AIsMergedPartner,
                 out Boolean AUserCanAccessPartner)
             {
-                return TRemote.MPartner.Partner.ServerLookups.WebConnectors.VerifyPartner(APartnerKey,
+                return TRemote.MPartner.Partner.ServerLookups.WebConnectors.VerifyPartnerAndGetDetails(APartnerKey,
                     out APartnerShortName,
                     out APartnerClass,
                     out AIsMergedPartner,
