--- before/csharp/ICT/Testing/lib/Common/Controls/ComboboxAutoPopulated.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Testing/lib/Common/Controls/ComboboxAutoPopulated.cs	2014-05-15 21:56:00.000000000 +0200
@@ -56,7 +56,7 @@
         [SetUp]
         public void Init()
         {
-            new TLogging("TestCommonControls.log");
+            new TLogging("../../log/TestCommonControls.log");
 
             TPetraConnector.Connect("../../etc/TestClient.config");
             FLedgerNumber = Convert.ToInt32(TAppSettingsManager.GetValue("LedgerNumber"));
