--- trunk/csharp/ICT/Petra/Client/app/Core/Cache.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Client/app/Core/Cache.cs	2014-05-10 06:53:13.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       christiank
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -28,7 +28,6 @@
 using System.Threading;
 using Ict.Petra.Shared.Interfaces.MFinance;
 using Ict.Petra.Shared.Interfaces.MPartner;
-using Ict.Petra.Shared.Interfaces.MConference;
 using Ict.Petra.Shared.MConference;
 using Ict.Petra.Shared.MFinance;
 using Ict.Petra.Shared.MPartner;
@@ -191,7 +190,7 @@
                 DataTable TmpDT;
 
                 // Refresh the Cacheble DataTable on the Serverside and return it
-                TRemote.MPartner.Subscriptions.Cacheable.WebConnectors.RefreshCacheableTable(ACacheableTable, out TmpDT);
+                TmpDT = TRemote.MPartner.Subscriptions.Cacheable.WebConnectors.RefreshCacheableTableAndGet(ACacheableTable);
                 UCacheableTablesManager.AddOrRefreshCachedTable(TmpDT, -1);
 
                 // Update the cached DataTable file
@@ -227,7 +226,7 @@
                 DataTable TmpDT;
 
                 // Refresh the Cacheble DataTable on the Serverside and return it
-                TRemote.MPartner.Mailing.Cacheable.WebConnectors.RefreshCacheableTable(ACacheableTable, out TmpDT);
+                TmpDT = TRemote.MPartner.Mailing.Cacheable.WebConnectors.RefreshCacheableTableAndGet(ACacheableTable);
                 UCacheableTablesManager.AddOrRefreshCachedTable(TmpDT, -1);
 
                 // Update the cached DataTable file
@@ -246,7 +245,7 @@
                 DataTable TmpDT;
 
                 // Refresh the Cacheble DataTable on the Serverside and return it
-                TRemote.MPartner.Partner.Cacheable.WebConnectors.RefreshCacheableTable(ACacheableTable, out TmpDT);
+                TmpDT = TRemote.MPartner.Partner.Cacheable.WebConnectors.RefreshCacheableTableAndGet(ACacheableTable);
                 UCacheableTablesManager.AddOrRefreshCachedTable(TmpDT, -1);
 
                 // Update the cached DataTable file
@@ -388,7 +387,7 @@
                 DataTable TmpDT;
 
                 // Refresh the Cacheble DataTable on the Serverside and return it
-                TRemote.MFinance.Cacheable.WebConnectors.RefreshCacheableTable(ACacheableTable, out TmpDT);
+                TmpDT = TRemote.MFinance.Cacheable.WebConnectors.RefreshCacheableTableAndGet(ACacheableTable);
                 UCacheableTablesManager.AddOrRefreshCachedTable(TmpDT, -1);
 
                 // Update the cached DataTable file
@@ -577,7 +576,7 @@
                 DataTable TmpDT;
 
                 // Refresh the Cacheable DataTable on the Serverside and return it
-                TRemote.MSysMan.Cacheable.WebConnectors.RefreshCacheableTable(ACacheableTable, out TmpDT);
+                TmpDT = TRemote.MSysMan.Cacheable.WebConnectors.RefreshCacheableTableAndGet(ACacheableTable);
                 UCacheableTablesManager.AddOrRefreshCachedTable(TmpDT, -1);
 
                 // Update the cached DataTable file
