--- before/csharp/ICT/Petra/Server/lib/MSysMan/TableMaintenance.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Server/lib/MSysMan/TableMaintenance.cs	2014-05-15 21:56:00.000000000 +0200
@@ -37,7 +37,7 @@
     /// <summary>
     /// User Interface Connector for table maintenance
     /// </summary>
-    public class TSysManTableMaintenanceUIConnector : TConfigurableMBRObject, ISysManUIConnectorsTableMaintenance
+    public class TSysManTableMaintenanceUIConnector : ISysManUIConnectorsTableMaintenance
     {
         /// <summary>
         /// Constructor
