--- before/csharp/ICT/Testing/exe/Controls/TextBoxNumberFormatting/Program.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Testing/exe/Controls/TextBoxNumberFormatting/Program.cs	2014-05-15 21:56:00.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       christiank
 //
-// Copyright 2004-2011 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -44,7 +44,7 @@
             Application.SetCompatibleTextRenderingDefault(true);
 
 
-            new TLogging("TextBoxNumberFormatting.log");
+            new TLogging("../../log/TextBoxNumberFormatting.log");
             Application.Run(new MainForm());
         }
     }
