--- trunk/csharp/ICT/Petra/Client/app/Core/Remoteobjects/RemoteObjects.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Client/app/Core/Remoteobjects/RemoteObjects.cs	2014-05-10 06:53:13.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       christiank, timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -22,24 +22,14 @@
 // along with OpenPetra.org.  If not, see <http://www.gnu.org/licenses/>.
 //
 using System;
-using System.Runtime.Remoting;
 using Ict.Common;
 using Ict.Common.Remoting.Shared;
-using Ict.Common.Remoting.Client;
-using Ict.Petra.Shared.Interfaces.MCommon;
-using Ict.Petra.Shared.Interfaces.MConference;
-using Ict.Petra.Shared.Interfaces.MPartner;
-using Ict.Petra.Shared.Interfaces.MPersonnel;
-using Ict.Petra.Shared.Interfaces.MSysMan;
-using Ict.Petra.Shared.Interfaces.MFinance;
-using Ict.Petra.Shared.Interfaces.MReporting;
+using Ict.Common.Remoting.Server;
 
 namespace Ict.Petra.Client.App.Core.RemoteObjects
 {
     /// <summary>
-    /// Holds all references to instantiated Serverside objects that are remoted by the Server.
-    /// These objects can get remoted either statically (the same Remoting URL all
-    /// the time) or dynamically (on-the-fly generation of Remoting URL).
+    /// Holds all references to objects that communicate with the server
     ///
     /// The TRemote class is used by the Client for all communication with the Server
     /// after the initial communication at Client start-up is done.
@@ -47,10 +37,10 @@
     /// The properties MPartner, MFinance, etc. represent the top-most level of the
     /// Petra Partner, Finance, etc. Petra Module Namespaces in the PetraServer.
     /// </summary>
-    public class TRemote : TRemoteBase
+    public class TRemote
     {
         /// <summary>Reference to the topmost level of the Petra Common Module Namespace</summary>
-        public static IMCommonNamespace MCommon
+        public static TMCommonNamespace MCommon
         {
             get
             {
@@ -59,7 +49,7 @@
         }
 
         /// <summary>Reference to the topmost level of the Petra Partner Module Namespace</summary>
-        public static IMConferenceNamespace MConference
+        public static TMConferenceNamespace MConference
         {
             get
             {
@@ -68,7 +58,7 @@
         }
 
         /// <summary>Reference to the topmost level of the Petra Partner Module Namespace</summary>
-        public static IMPartnerNamespace MPartner
+        public static TMPartnerNamespace MPartner
         {
             get
             {
@@ -77,7 +67,7 @@
         }
 
         /// <summary>Reference to the topmost level of the Petra Personnel Module Namespace</summary>
-        public static IMPersonnelNamespace MPersonnel
+        public static TMPersonnelNamespace MPersonnel
         {
             get
             {
@@ -86,7 +76,7 @@
         }
 
         /// <summary>Reference to  the topmost level of the Petra Finance Module Namespace</summary>
-        public static IMFinanceNamespace MFinance
+        public static TMFinanceNamespace MFinance
         {
             get
             {
@@ -95,7 +85,7 @@
         }
 
         /// <summary>Reference to  the topmost level of the Petra Reporting Module Namespace</summary>
-        public static IMReportingNamespace MReporting
+        public static TMReportingNamespace MReporting
         {
             get
             {
@@ -104,7 +94,7 @@
         }
 
         /// <summary>Reference to the topmost level of the Petra System Manager Module Namespace</summary>
-        public static IMSysManNamespace MSysMan
+        public static TMSysManNamespace MSysMan
         {
             get
             {
@@ -112,43 +102,26 @@
             }
         }
 
-        private static IMCommonNamespace UCommonObject;
-        private static IMConferenceNamespace UConferenceObject;
-        private static IMPartnerNamespace UPartnerObject;
-        private static IMPersonnelNamespace UPersonnelObject;
-        private static IMFinanceNamespace UFinanceObject;
-        private static IMReportingNamespace UReportingObject;
-        private static IMSysManNamespace USysManObject;
+        private static TMCommonNamespace UCommonObject;
+        private static TMConferenceNamespace UConferenceObject;
+        private static TMPartnerNamespace UPartnerObject;
+        private static TMPersonnelNamespace UPersonnelObject;
+        private static TMFinanceNamespace UFinanceObject;
+        private static TMReportingNamespace UReportingObject;
+        private static TMSysManNamespace USysManObject;
 
         /// <summary>
-        /// References to the rest of the topmost level of the other Petra Module Namespaces will go here
+        /// initialize the references
         /// </summary>
-        /// <returns>void</returns>
-        public TRemote(IClientManagerInterface AClientManager,
-            IMCommonNamespace ACommonObject,
-            IMConferenceNamespace AConferenceObject,
-            IMPartnerNamespace APartnerObject,
-            IMPersonnelNamespace APersonnelObject,
-            IMFinanceNamespace AFinanceObject,
-            IMReportingNamespace AReportingObject,
-            IMSysManNamespace ASysManObject)
-            : base(AClientManager)
-        {
-            USysManObject = ASysManObject;
-            UConferenceObject = AConferenceObject;
-            UPartnerObject = APartnerObject;
-            UCommonObject = ACommonObject;
-            UPersonnelObject = APersonnelObject;
-            UFinanceObject = AFinanceObject;
-            UReportingObject = AReportingObject;
-
-            FRemoteObjects.Add((MarshalByRefObject)USysManObject);
-            FRemoteObjects.Add((MarshalByRefObject)UConferenceObject);
-            FRemoteObjects.Add((MarshalByRefObject)UPartnerObject);
-            FRemoteObjects.Add((MarshalByRefObject)UCommonObject);
-            FRemoteObjects.Add((MarshalByRefObject)UPersonnelObject);
-            FRemoteObjects.Add((MarshalByRefObject)UFinanceObject);
-            FRemoteObjects.Add((MarshalByRefObject)UReportingObject);
+        public TRemote()
+        {
+            USysManObject = new TMSysManNamespace();
+            UConferenceObject = new TMConferenceNamespace();
+            UPartnerObject = new TMPartnerNamespace();
+            UCommonObject = new TMCommonNamespace();
+            UPersonnelObject = new TMPersonnelNamespace();
+            UFinanceObject = new TMFinanceNamespace();
+            UReportingObject = new TMReportingNamespace();
         }
     }
 }
\ No newline at end of file
