--- trunk/csharp/ICT/Petra/Server/lib/MConference/ConferenceOptions.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Server/lib/MConference/ConferenceOptions.cs	2014-05-10 06:53:13.000000000 +0200
@@ -2,9 +2,9 @@
 // DO NOT REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 //
 // @Authors:
-//       berndr
+//       berndr, timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -315,8 +315,7 @@
         /// <param name="AUnitKey">Partner Key of the unit from which the outreach options are retrieved</param>
         /// <param name="AConferenceTable">A table with all the units</param>
         /// <returns></returns>
-        [RequireModulePermission("CONFERENCE")]
-        public static System.Boolean GetOutreachOptions(long AUnitKey,
+        private static System.Boolean GetOutreachOptions(long AUnitKey,
             out System.Data.DataTable AConferenceTable)
         {
             AConferenceTable = new DataTable();
