--- trunk/csharp/ICT/Petra/Server/lib/MCommon/FieldOfService.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Server/lib/MCommon/FieldOfService.cs	2014-05-10 06:53:13.000000000 +0200
@@ -59,7 +59,7 @@
     ///          However, Server Objects that derive from these objects and that
     ///          are also UIConnectors are feasible.
     /// </summary>
-    public class TFieldOfServiceUIConnector : TConfigurableMBRObject, IPartnerUIConnectorsFieldOfService
+    public class TFieldOfServiceUIConnector : IPartnerUIConnectorsFieldOfService
     {
         /// <summary>Name of the Main DataSet for the Screen</summary>
         private const String DATASETNAME = "FieldOfServiceScreen";
