--- before/csharp/ICT/Petra/Server/lib/MPartner/Common/ImportPartnerForm.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Server/lib/MPartner/Common/ImportPartnerForm.cs	2014-05-15 21:56:00.000000000 +0200
@@ -33,6 +33,7 @@
 using Ict.Common.DB;
 using Ict.Common.IO;
 using Ict.Common.Verification;
+using Ict.Common.Remoting.Server;
 using Ict.Petra.Server.MCommon.Data.Cascading;
 using Ict.Petra.Server.App.Core;
 using Ict.Petra.Server.MPartner.Common;
