--- before/csharp/ICT/Petra/Client/MSysMan/Gui/IntranetExportMotivationsDialog.ManualCode.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Client/MSysMan/Gui/IntranetExportMotivationsDialog.ManualCode.cs	2014-05-15 21:56:00.000000000 +0200
@@ -79,7 +79,7 @@
             /* load available ledgers into listbox */
             DataTable LedgerTable;
 
-            TRemote.MFinance.Cacheable.WebConnectors.RefreshCacheableTable(TCacheableFinanceTablesEnum.LedgerNameList, out LedgerTable);
+            LedgerTable = TRemote.MFinance.Cacheable.WebConnectors.RefreshCacheableTableAndGet(TCacheableFinanceTablesEnum.LedgerNameList);
 
             foreach (DataRow Row in LedgerTable.Rows)
             {
