--- trunk/csharp/ICT/Petra/Server/lib/MSysMan/UserManager.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Server/lib/MSysMan/UserManager.cs	2014-05-10 06:53:13.000000000 +0200
@@ -225,16 +225,16 @@
         /// </summary>
         /// <param name="AUserID"></param>
         /// <param name="APassword"></param>
-        /// <param name="AProcessID"></param>
         /// <param name="ASystemEnabled"></param>
         /// <returns></returns>
         [NoRemoting]
-        public static TPetraPrincipal PerformUserAuthentication(String AUserID, String APassword, out Int32 AProcessID, out Boolean ASystemEnabled)
+        public static TPetraPrincipal PerformUserAuthentication(String AUserID, String APassword, out Boolean ASystemEnabled)
         {
             DateTime LoginDateTime;
             TPetraPrincipal PetraPrincipal = null;
 
-            AProcessID = -1;
+            Int32 AProcessID = -1;
+
             ASystemEnabled = true;
 
             string EmailAddress = AUserID;
@@ -493,10 +493,12 @@
         [RequireModulePermission("NONE")]
         public static void SignalReloadCachedUserInfo(String AUserID)
         {
-            Ict.Petra.Server.App.Core.DomainManager.ClientTaskAddToOtherClient(AUserID,
+            TClientManager.QueueClientTask(AUserID,
                 SharedConstants.CLIENTTASKGROUP_USERINFOREFRESH,
                 "",
-                1);
+                null, null, null, null,
+                1,
+                -1);
         }
     }
 }
@@ -524,10 +526,9 @@
         /// authenticate a user
         /// </summary>
         public IPrincipal PerformUserAuthentication(string AUserName, string APassword,
-            out Int32 AProcessID,
             out Boolean ASystemEnabled)
         {
-            return TUserManagerWebConnector.PerformUserAuthentication(AUserName, APassword, out AProcessID, out ASystemEnabled);
+            return TUserManagerWebConnector.PerformUserAuthentication(AUserName, APassword, out ASystemEnabled);
         }
     }
 }
\ No newline at end of file
