--- trunk/csharp/ICT/Petra/Client/MFinance/Gui/AP/APMain.ManualCode.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Client/MFinance/Gui/AP/APMain.ManualCode.cs	2014-05-10 06:53:13.000000000 +0200
@@ -179,40 +179,38 @@
             // Check whether this thread should still execute
             while (FKeepUpSearchFinishedCheck)
             {
-                TAsyncExecProgressState ThreadStatus;
+                TProgressState ThreadStatus;
 
+                /* The next line of code calls a function on the PetraServer
+                 * > causes a bit of data traffic everytime! */
                 if (FSearchForSuppliers)
                 {
-                    ThreadStatus = FSupplierFindObject.AsyncExecProgress.ProgressState;
+                    ThreadStatus = FSupplierFindObject.Progress;
                 }
                 else
                 {
-                    ThreadStatus = FInvoiceFindObject.AsyncExecProgress.ProgressState;
+                    ThreadStatus = FInvoiceFindObject.Progress;
                 }
 
-                /* The next line of code calls a function on the PetraServer
-                 * > causes a bit of data traffic everytime! */
-                switch (ThreadStatus)
+                if (ThreadStatus.JobFinished)
                 {
-                    case TAsyncExecProgressState.Aeps_Finished:
-                        FKeepUpSearchFinishedCheck = false;
+                    FKeepUpSearchFinishedCheck = false;
 
-                        // see also http://stackoverflow.com/questions/6184/how-do-i-make-event-callbacks-into-my-win-forms-thread-safe
-                        if (InvokeRequired)
-                        {
-                            Invoke(new SimpleDelegate(FinishThread));
-                        }
-                        else
-                        {
-                            FinishThread();
-                        }
-
-                        break;
-
-                    case TAsyncExecProgressState.Aeps_Stopped:
-                        FKeepUpSearchFinishedCheck = false;
-                        EnableDisableUI(true);
-                        return;
+                    // see also http://stackoverflow.com/questions/6184/how-do-i-make-event-callbacks-into-my-win-forms-thread-safe
+                    if (InvokeRequired)
+                    {
+                        Invoke(new SimpleDelegate(FinishThread));
+                    }
+                    else
+                    {
+                        FinishThread();
+                    }
+                }
+                else if (ThreadStatus.CancelJob)
+                {
+                    FKeepUpSearchFinishedCheck = false;
+                    EnableDisableUI(true);
+                    return;
                 }
 
                 // Sleep a bit, then loop...
