--- trunk/csharp/ICT/Petra/Server/lib/MPartner/connect/Partner.PartnerLocationFind.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Server/lib/MPartner/connect/Partner.PartnerLocationFind.cs	2014-05-10 06:53:13.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       timh, timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -48,20 +48,17 @@
     /// Partner Location Search Screen UIConnector
     ///
     /// </summary>
-    public class TPartnerLocationFindUIConnector : TConfigurableMBRObject, IPartnerUIConnectorsPartnerLocationFind
+    public class TPartnerLocationFindUIConnector : IPartnerUIConnectorsPartnerLocationFind
     {
-        private TAsynchronousExecutionProgress FAsyncExecProgress;
         private Thread FFindThread;
         private TPagedDataSet FPagedDataSetObject;
 
-        /// <summary>Property accessor Returns reference to the Asynchronous execution control object to the caller</summary>
-        public IAsynchronousExecutionProgress AsyncExecProgress
+        /// <summary>Get the current state of progress</summary>
+        public TProgressState Progress
         {
             get
             {
-                return (IAsynchronousExecutionProgress)TCreateRemotableObject.CreateRemotableObject(
-                    typeof(TAsynchronousExecutionProgressRemote),
-                    FAsyncExecProgress);
+                return FPagedDataSetObject.Progress;
             }
         }
 
@@ -77,9 +74,7 @@
             OdbcParameter miParam;
             DataRow CriteriaRow;
 
-            FAsyncExecProgress = new TAsynchronousExecutionProgress();
             FPagedDataSetObject = new TPagedDataSet(new PartnerFindTDSSearchResultTable());
-            FPagedDataSetObject.AsyncExecProgress = FAsyncExecProgress;
             ColumnNameMapping = null;
 
             // get the first and only row
@@ -174,6 +169,7 @@
             {
                 ThreadStart ThreadStartDelegate = new ThreadStart(FPagedDataSetObject.ExecuteQuery);
                 FFindThread = new Thread(ThreadStartDelegate);
+                FFindThread.Name = "PartnerLocationFind" + Guid.NewGuid().ToString();
                 FFindThread.Start();
             }
             catch (Exception)
