--- before/csharp/ICT/Petra/Server/app/Core/ModuleAccessManager.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Petra/Server/app/Core/ModuleAccessManager.cs	2014-05-15 21:56:00.000000000 +0200
@@ -326,4 +326,17 @@
         /// </summary>
         public readonly string RequiredModulesExpression;
     }
+
+    /// <summary>
+    /// attribute for annotation of server functions. Instantiator will check access permissions
+    /// </summary>
+    public class CheckServerAdminTokenAttribute : System.Attribute
+    {
+        /// <summary>
+        /// constructor
+        /// </summary>
+        public CheckServerAdminTokenAttribute()
+        {
+        }
+    }
 }
\ No newline at end of file
