--- before/csharp/ICT/Common/Remoting/Client/KeepAlive.cs	2014-05-15 21:53:13.000000000 +0200
+++ patched/csharp/ICT/Common/Remoting/Client/KeepAlive.cs	2014-05-15 21:56:00.000000000 +0200
@@ -69,6 +69,7 @@
 
             // Start KeepAliveThread
             TheThread = new Thread(new ThreadStart(KeepAliveThread));
+            TheThread.Name = "KeepAliveThread" + Guid.NewGuid().ToString();
             TheThread.Start();
         }
 
@@ -135,7 +136,7 @@
             {
                 Register((IKeepAlive)ARemotedObject);
             }
-            else
+            else if (!TClientSettings.RunAsStandalone)
             {
                 TLogging.Log("KeepAlive Register: " + ARemotedObject.GetType().ToString() + " does not implement IKeepAlive");
             }
