--- trunk/csharp/ICT/Petra/Shared/Security.Principal.cs	2014-05-10 06:50:27.000000000 +0200
+++ remoting/csharp/ICT/Petra/Shared/Security.Principal.cs	2014-05-10 06:53:13.000000000 +0200
@@ -23,8 +23,8 @@
 //
 using System;
 using System.Data;
-using System.Security.Principal;
 using System.Runtime.Serialization;
+using System.Security.Principal;
 using Ict.Common;
 using Ict.Common.Exceptions;
 using Ict.Petra.Shared.Security;
