--- before/i18n/i18n.build	2014-05-15 21:53:13.000000000 +0200
+++ patched/i18n/i18n.build	2014-05-15 21:56:00.000000000 +0200
@@ -31,16 +31,20 @@
 </target>
 
 <target name="downloadTranslations">
-  <exec program="bzr" commandline="checkout http://bazaar.launchpad.net/~jomammele/openpetraorg/translationfiles" workingdir="${tempdir}"/>
-  <move todir="${dir.root}/i18n/">
-    <fileset basedir="${tempdir}/translationfiles/i18n">
-      <include name="*.po" />
-    </fileset>
-  </move>
-  <delete dir="${tempdir}/translationfiles"/>
-  <move file="${dir.root}/i18n/de.po" tofile="${dir.root}/i18n/de-DE.po"/>
-  <move file="${dir.root}/i18n/da.po" tofile="${dir.root}/i18n/da-DK.po"/>
-  <move file="${dir.root}/i18n/es.po" tofile="${dir.root}/i18n/es-ES.po"/>
+  <property name="OpenBuildService" value="false" overwrite="false"/>
+  <!-- on open build service, we cannot download any files. use the translation files from the git, or add to download _services -->
+  <if test="${OpenBuildService == 'false'}">
+    <exec program="bzr" commandline="checkout http://bazaar.launchpad.net/~jomammele/openpetraorg/translationfiles" workingdir="${tempdir}"/>
+    <move todir="${dir.root}/i18n/">
+      <fileset basedir="${tempdir}/translationfiles/i18n">
+        <include name="*.po" />
+      </fileset>
+    </move>
+    <delete dir="${tempdir}/translationfiles"/>
+    <move file="${dir.root}/i18n/de.po" tofile="${dir.root}/i18n/de-DE.po"/>
+    <move file="${dir.root}/i18n/da.po" tofile="${dir.root}/i18n/da-DK.po"/>
+    <move file="${dir.root}/i18n/es.po" tofile="${dir.root}/i18n/es-ES.po"/>
+  </if>
 </target>
 
 <target name="compile" depends="translation" 
