diff --git a/OpenPetra.build b/OpenPetra.build
index 4fbd7cd..d4bc6a6 100644
--- a/OpenPetra.build
+++ b/OpenPetra.build
@@ -22,14 +22,16 @@ list of available commands:
   compileProject: compile a single project file
       nant compileProject -D:file=delivery/projects/sharpdevelop4/Ict.Testing.ControlTestBench.csproj 
   start: start both the local server and client
-  startPetraServer: run a local Petra Server
-  stopPetraServer: stop the local Petra Server
-  startPetraClient: run the Petra Client
+  startServer: run a local OpenPetra Server
+  stopServer: stop the local OpenPetra Server
+  startClient: run the OpenPetra Client
+  startStandalone: run the OpenPetra standalone application, if generateSolution was run with -D:compileForStandalone=true
   deleteBakFiles: delete all backup files
   initConfigFiles: write and overwrite the config files for client, server and testing, and the version.txt
-  test: will run the NUnit tests
+  test: will run the NUnit tests (smaller tests: test-without-display, test-main-navigation-screens, test-client)
       nant test-without-display             runs all basic tests (excl. GUI tests)
       nant test-main-navigation-screens     tests the opening of all screens from the main menu
+      nant test-client
   devzip: build a zip file for developers for easy start
 
 compat:
@@ -114,7 +116,7 @@ useful parameters
   <echo message="Target translation started at: ${datetime::now()}" /> 
   
   <property name="target" value="compile" overwrite="false" />
-  <nant inheritall="false" target="${target}" 
+  <nant inheritall="true" target="${target}" 
         buildfile="i18n/i18n.build" />
         
   <echo message="Target translation finished at: ${datetime::now()}" /> 
@@ -285,6 +287,7 @@ useful parameters
     CodeRootDir="${dir.csharp}"
     NamespaceMapFilename="${namespaceFile}"
     NamespaceMap3rdParty="${namespaceFile3rdParty}"
+    CompilingForStandalone="${compileForStandalone}"
     DependencyMapFilename="${projectDependenciesFile}"/>
     
   <echo message="Target generateNamespaceMap finished at: ${datetime::now()}" />         
@@ -301,7 +304,7 @@ useful parameters
     NamespaceMap3rdParty="${namespaceFile3rdParty}"
     DependencyMapFilename="${projectDependenciesFile}"
     ShowWarnings="false"
-    LimitToNamespaces="Ict.Common,Ict.Common.IO,Ict.Common.DB,Ict.Tools*"/>
+    LimitToNamespaces="Ict.Common,Ict.Common.IO,Ict.Common.DB,Ict.Common.Session,Ict.Tools*"/>
   <nant inheritall="false" target="compile" 
         buildfile="csharp/ThirdParty/ThirdParty.build" />
   <call target="generateProjectFilesInternal"/>
@@ -441,7 +444,7 @@ useful parameters
   <echo message="Target previewWinform (outer) finished at: ${datetime::now()}" />
 </target>
 
-<target name="test" depends="initConfigFiles">
+<target name="test" depends="initConfigFiles, set-server-port-for-jenkins">
   <echo message="Target test started at: ${datetime::now()}" />
   
   <loadtasks assembly="${Ict.Tools.NAntTasks.DLL}" unless="${task::exists('GenerateNamespaceMap')}"/>
@@ -450,7 +453,7 @@ useful parameters
   <echo message="Target test finished at: ${datetime::now()}" />
 </target>
 
-<target name="test-without-display" depends="initConfigFiles">
+<target name="test-without-display" depends="initConfigFiles, set-server-port-for-jenkins">
   <echo message="Target test-without-display (outer) started at: ${datetime::now()}" />
   
   <loadtasks assembly="${Ict.Tools.NAntTasks.DLL}" unless="${task::exists('GenerateNamespaceMap')}"/>
@@ -459,7 +462,7 @@ useful parameters
   <echo message="Target test-without-display (outer) finished at: ${datetime::now()}" />
 </target>
 
-<target name="test-client" depends="initConfigFiles">
+<target name="test-client" depends="initConfigFiles, set-server-port-for-jenkins">
   <echo message="Target test-client (outer) started at: ${datetime::now()}" />
   
   <loadtasks assembly="${Ict.Tools.NAntTasks.DLL}" unless="${task::exists('GenerateNamespaceMap')}"/>
@@ -468,7 +471,7 @@ useful parameters
   <echo message="Target test-client (outer) finished at: ${datetime::now()}" />
 </target>
 
-<target name="test-main-navigation-screens-core" depends="initConfigFiles">
+<target name="test-main-navigation-screens-core" depends="initConfigFiles, set-server-port-for-jenkins">
   <echo message="Target test-main-navigation-screens-core (outer) started at: ${datetime::now()}" />
   
   <loadtasks assembly="${Ict.Tools.NAntTasks.DLL}" unless="${task::exists('GenerateNamespaceMap')}"/>
@@ -477,7 +480,7 @@ useful parameters
   <echo message="Target test-main-navigation-screens-core (outer) finished at: ${datetime::now()}" />
 </target>
 
-<target name="test-main-navigation-screens" depends="initConfigFiles">
+<target name="test-main-navigation-screens" depends="initConfigFiles, set-server-port-for-jenkins">
   <echo message="Target test-main-navigation-screens (outer) started at: ${datetime::now()}" />
   
   <loadtasks assembly="${Ict.Tools.NAntTasks.DLL}" unless="${task::exists('GenerateNamespaceMap')}"/>
@@ -493,12 +496,31 @@ useful parameters
 &lt;project name="OpenPetra-userconfig">
     &lt;property name="DBMS.Type" value="sqlite"/>
     &lt;property name="DBMS.Password" value=""/>
+    &lt;property name="Server.Port" value="${Server.Port}"/>
 &lt;/project>
     </echo>
     
     <echo message="Target enable-sqlite-config-for-jenkins finished at: ${datetime::now()}" />
 </target>
 
+<!-- need to set the server port here, since it is not passed when called the Unit tests, so they are not using the server port defined on the command line;
+     only applies if no local OpenPetra.build.config file exists -->
+<target name="set-server-port-for-jenkins">
+    <if test="${not file::exists('OpenPetra.build.config') and not file::exists('OpenPetra.build.config.my')}">
+        <echo file="OpenPetra.build.config">&lt;?xml version="1.0"?>
+    &lt;project name="OpenPetra-userconfig">
+        &lt;property name="Server.Port" value="${Server.Port}"/>
+        &lt;property name="PostgreSQL.Version" value="${PostgreSQL.Version}"/>
+        &lt;property name="DBMS.Type" value="${DBMS.Type}"/>
+        &lt;property name="DBMS.DBHostOrFile" value="${DBMS.DBHostOrFile}"/>
+        &lt;property name="DBMS.DBName" value="${DBMS.DBName}"/>
+        &lt;property name="DBMS.UserName" value="${DBMS.UserName}"/>
+        &lt;property name="DBMS.Password" value="${DBMS.Password}"/>
+    &lt;/project>
+        </echo>
+    </if>
+</target>
+
 <target name="devzip">
     <echo message="Target devzip started at: ${datetime::now()}" />
     
@@ -544,5 +566,6 @@ useful parameters
     
     <echo message="Target devzip finished at: ${datetime::now()}" />
 </target>
+
 </project>
 
