--- before/setup/petra0300/linuxclient/startClient.sh	2014-05-15 21:53:13.000000000 +0200
+++ patched/setup/petra0300/linuxclient/startClient.sh	2014-05-15 21:56:00.000000000 +0200
@@ -1,8 +1,15 @@
 #!/bin/bash
 cd bin30
-if [ ! -f ../etc30/publickey.xml ]
+. /opt/mono-openpetra/env.sh
+if [ ! -z "$1" ]
 then
-  cp ../etc30/publickey-sample.xml ../etc30/publickey.xml
+	# replace hostname in config file; need to run as root with permissions
+	if [ -w ../etc30/PetraClientRemote.config ]
+	then
+		sed -i "s#http://.*9000#http://$1/openpetra9000#g" ../etc30/PetraClientRemote.config
+	fi
+	mono PetraClient.exe -C:../etc30/PetraClientRemote.config -OpenPetra.HTTPServer:http://$1/openpetra9000 &
+else
+	mono PetraClient.exe -C:../etc30/PetraClientRemote.config &
 fi
-mono PetraClient.exe -C:../etc30/PetraClientRemote.config &
 cd ..
