--- openpetraNow.freeze2014_05/csharp/ICT/Testing/lib/Reporting/ReportTesting.tools.cs	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Testing/lib/Reporting/ReportTesting.tools.cs	2014-05-15 22:04:07.000000000 +0200
@@ -113,7 +113,7 @@
 
             ReportGenerator.Start(Parameters.ToDataTable());
 
-            while (ReportGenerator.AsyncExecProgressServerSide.ProgressState == TAsyncExecProgressState.Aeps_Executing)
+            while (!ReportGenerator.Progress.JobFinished)
             {
                 Thread.Sleep(500);
             }
