--- openpetraNow.freeze2014_05/csharp/ICT/Petra/Client/app/MainWindow/MainWindowNew.ManualCode.cs	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Petra/Client/app/MainWindow/MainWindowNew.ManualCode.cs	2014-05-15 22:04:07.000000000 +0200
@@ -68,7 +68,9 @@
 
         private static bool FConferenceSelected = false;
         private static Int64 FConferenceKey = 0;
+#if TODORemoting
         private PetraClient_AutomatedAppTest.TAutomatedAppTest TestRunner;
+#endif
 
         /// <summary>
         /// The currently selected Ledger
@@ -144,6 +146,7 @@
         {
             string DisconnectTimeFromCommandLine = TAppSettingsManager.GetValue("DisconnectTime");
 
+#if TODORemoting
             if (TAppSettingsManager.GetBoolean("RunAutoTests", false) == true)
             {
                 // We need to manually 'fix up' the value of DisconnectTime that we get from .NET when we request
@@ -165,7 +168,10 @@
                 TestRunner.ClientID = TConnectionManagementBase.GConnectionManagement.ClientID;
                 TestRunner.Start(this);
             }
-            else if (System.Windows.Forms.Form.ModifierKeys != Keys.Control)
+            else
+#endif
+
+            if (System.Windows.Forms.Form.ModifierKeys != Keys.Control)
             {
                 string testAction = TAppSettingsManager.GetValue("TestAction");
 
