--- openpetra.freeze/csharp/ICT/Petra/Server/lib/MPartner/web/Partner.cs	2014-07-24 14:18:45.000000000 +0200
+++ openpetra.patched/csharp/ICT/Petra/Server/lib/MPartner/web/Partner.cs	2014-07-24 14:19:19.000000000 +0200
@@ -29,6 +29,7 @@
 using Ict.Common.Verification;
 using Ict.Common.Verification.Exceptions;
 using Ict.Common.DB;
+using Ict.Common.Remoting.Server;
 using Ict.Petra.Shared;
 using Ict.Petra.Shared.MFinance.Account.Data;
 using Ict.Petra.Shared.MFinance.Gift.Data;
