--- openpetraNow.freeze2014_05/csharp/ICT/Petra/Server/lib/MPartner/connect/Partner.PartnerEdit.cs	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Petra/Server/lib/MPartner/connect/Partner.PartnerEdit.cs	2014-05-15 22:04:07.000000000 +0200
@@ -86,7 +86,7 @@
     ///          are also UIConnectors are feasible.
     ///
     /// </summary>
-    public partial class TPartnerEditUIConnector : TConfigurableMBRObject, IPartnerUIConnectorsPartnerEdit
+    public partial class TPartnerEditUIConnector : IPartnerUIConnectorsPartnerEdit
     {
         private const String DATASETNAME = "PartnerEditScreen";
 
@@ -1985,7 +1985,9 @@
 
                 TVerificationResultCollection SingleVerificationResultCollection;
                 AVerificationResult = new TVerificationResultCollection();
-                TDBTransaction SubmitChangesTransaction = DBAccess.GDBAccessObj.BeginTransaction(IsolationLevel.Serializable);
+                bool NewTransaction;
+                TDBTransaction SubmitChangesTransaction = DBAccess.GDBAccessObj.GetNewOrExistingTransaction(IsolationLevel.Serializable,
+                    out NewTransaction);
 
                 PrepareBankingDetailsForSaving(ref AInspectDS, ref AVerificationResult, SubmitChangesTransaction);
 
@@ -2116,7 +2118,11 @@
                         // Must call AcceptChanges so that DataSet.Merge on Client side works
                         // properly if Primary Keys have been changed!
                         AInspectDS.AcceptChanges();
-                        DBAccess.GDBAccessObj.CommitTransaction();
+
+                        if (NewTransaction)
+                        {
+                            DBAccess.GDBAccessObj.CommitTransaction();
+                        }
 
                         /* $IFDEF DEBUGMODE if TLogging.DL >= 9 then Console.WriteLine('Location[0] LocationKey: ' + FSubmissionDS.PLocation[0].LocationKey.ToString + '; PartnerLocation[0] LocationKey: ' +
                          *FSubmissionDS.PPartnerLocation[0].LocationKey.ToString);$ENDIF */
@@ -2124,7 +2130,11 @@
                     }
                     else
                     {
-                        DBAccess.GDBAccessObj.RollbackTransaction();
+                        if (NewTransaction)
+                        {
+                            DBAccess.GDBAccessObj.RollbackTransaction();
+                        }
+
                         TLogging.LogAtLevel(8, "TPartnerEditUIConnector.SubmitChanges: Transaction ROLLED BACK!");
                     }
                 }
