--- a/csharp/ICT/Petra/Server/lib/MPartner/connect/Partner.PartnerEdit.cs
+++ b/csharp/ICT/Petra/Server/lib/MPartner/connect/Partner.PartnerEdit.cs
@@ -86,7 +86,7 @@ namespace Ict.Petra.Server.MPartner.Partner.UIConnectors
     ///          are also UIConnectors are feasible.
     ///
     /// </summary>
-    public partial class TPartnerEditUIConnector : TConfigurableMBRObject, IPartnerUIConnectorsPartnerEdit
+    public partial class TPartnerEditUIConnector : IPartnerUIConnectorsPartnerEdit
     {
         private const String DATASETNAME = "PartnerEditScreen";
 
