--- openpetra.freeze/csharp/ICT/Petra/Client/MPartner/Gui/PartnerEdit.ManualCode.cs	2014-07-24 14:18:45.000000000 +0200
+++ openpetra.patched/csharp/ICT/Petra/Client/MPartner/Gui/PartnerEdit.ManualCode.cs	2014-07-24 14:21:43.000000000 +0200
@@ -3311,11 +3311,6 @@
                 }
             } while (!(ServerCallSuccessful));
 
-            if (ServerCallSuccessful)
-            {
-                // Register Object with the TEnsureKeepAlive Class so that it doesn't get GC'd
-                TEnsureKeepAlive.Register(FPartnerEditUIConnector);
-            }
 
             return ServerCallSuccessful;
         }
