--- openpetraNow.freeze2014_05/csharp/ICT/Testing/lib/Common/Controls/ComboboxAutoPopulated.cs	2014-07-24 12:24:11.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Testing/lib/Common/Controls/ComboboxAutoPopulated.cs	2014-07-24 13:29:04.000000000 +0200
@@ -56,7 +56,7 @@
         [SetUp]
         public void Init()
         {
-            new TLogging("TestCommonControls.log");
+            new TLogging("../../log/TestCommonControls.log");
 
             TPetraConnector.Connect("../../etc/TestClient.config");
             FLedgerNumber = Convert.ToInt32(TAppSettingsManager.GetValue("LedgerNumber"));
@@ -65,6 +65,7 @@
         /// <summary>
         /// clean up, disconnect from OpenPetra server
         /// </summary>
+        [TearDown]
         public void TearDown()
         {
             TPetraConnector.Disconnect();
