diff --git a/csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs b/csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs
index 959045b..87fd12c 100644
--- a/csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs
+++ b/csharp/ICT/Petra/Client/CommonDialogs/ProgressDialog.cs
@@ -38,6 +38,7 @@ namespace Ict.Petra.Client.CommonDialogs
     /// </summary>
     public partial class TProgressDialog : System.Windows.Forms.Form
     {
+        private Thread FWorkerThread = null;
         private bool FConfirmedClosing = false;
         private bool FShowCancellationConfirmationQuestion = false;
         private bool FCancelled = false;
@@ -74,7 +75,8 @@ namespace Ict.Petra.Client.CommonDialogs
             FShowCancellationConfirmationQuestion = AShowCancellationConfirmationQuestion;
 
             TRemote.MCommon.WebConnectors.Reset();
-            AWorkerThread.Start();
+            FWorkerThread = AWorkerThread;
+            FWorkerThread.Start();
 
             FQueryServerForProgress = AQueryServerForProgress;
 
@@ -113,19 +115,20 @@ namespace Ict.Petra.Client.CommonDialogs
 
             if (CancelConfirmationResult == DialogResult.Yes)
             {
+                FCancelled = true;
+
                 if (FQueryServerForProgress)
                 {
                     try
                     {
                         TRemote.MCommon.WebConnectors.CancelJob();
+                        FWorkerThread.Join();
                     }
                     catch (Exception Exc)
                     {
                         TLogging.Log("While cancelling a job from a Progress Dialog we got the following Exception:\r\n" + Exc.ToString());
                     }
                 }
-
-                FCancelled = true;
             }
         }
 
@@ -183,6 +186,8 @@ namespace Ict.Petra.Client.CommonDialogs
             }
         }
 
+        private int FCountConsecutiveErrors = 0;
+
         private void Timer1Tick(object sender, EventArgs e)
         {
             string caption;
@@ -192,20 +197,35 @@ namespace Ict.Petra.Client.CommonDialogs
 
             if (FQueryServerForProgress)
             {
-                if (TRemote.MCommon.WebConnectors.GetCurrentState(out caption,
-                        out message,
-                        out percentage,
-                        out finished))
+                try
+                {
+                    if (TRemote.MCommon.WebConnectors.GetCurrentState(out caption,
+                            out message,
+                            out percentage,
+                            out finished))
+                    {
+                        FCountConsecutiveErrors = 0;
+                        this.Text = caption;
+                        this.lblMessage.Text = message;
+                        this.progressBar.Value = percentage;
+
+                        if (finished)
+                        {
+                            // wait till the thread finishes
+                            FWorkerThread.Join();
+                            this.DialogResult = FCancelled ? DialogResult.Cancel : DialogResult.OK;
+                            FConfirmedClosing = true;
+                            Close();
+                        }
+                    }
+                }
+                catch (Exception ex)
                 {
-                    this.Text = caption;
-                    this.lblMessage.Text = message;
-                    this.progressBar.Value = percentage;
+                    FCountConsecutiveErrors++;
 
-                    if (finished)
+                    if (FCountConsecutiveErrors > 3)
                     {
-                        this.DialogResult = FCancelled ? DialogResult.Cancel : DialogResult.OK;
-                        FConfirmedClosing = true;
-                        Close();
+                        throw ex;
                     }
                 }
             }
@@ -221,6 +241,8 @@ namespace Ict.Petra.Client.CommonDialogs
 
                 if (FFinished)
                 {
+                    // wait till the thread finishes
+                    FWorkerThread.Join();
                     this.DialogResult = FCancelled ? DialogResult.Cancel : DialogResult.OK;
                     FConfirmedClosing = true;
                     Close();
@@ -237,4 +259,4 @@ namespace Ict.Petra.Client.CommonDialogs
             }
         }
     }
-}
\ No newline at end of file
+}
