--- openpetraNow.freeze2014_05/csharp/ICT/Testing/lib/Plugins/BankImportMT940/Test.MT940.cs	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Testing/lib/Plugins/BankImportMT940/Test.MT940.cs	2014-05-16 08:52:04.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -50,7 +50,7 @@
         [TestFixtureSetUp]
         public void Init()
         {
-            new TLogging("TestClient.log");
+            new TLogging("../../log/TestClient.log");
             new TAppSettingsManager("../../etc/TestClient.config");
         }
 
