--- openpetraNow.freeze2014_05/csharp/ICT/Petra/Server/lib/MFinance/GL/GL.Importing.cs	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Petra/Server/lib/MFinance/GL/GL.Importing.cs	2014-05-15 22:04:07.000000000 +0200
@@ -32,6 +32,7 @@
 using Ict.Common.DB;
 using Ict.Common.Exceptions;
 using Ict.Common.Verification;
+using Ict.Common.Remoting.Server;
 using Ict.Petra.Server.MFinance.Account.Data.Access;
 using Ict.Petra.Server.MPartner.Partner.Data.Access;
 using Ict.Petra.Server.MSysMan.Data.Access;
