--- openpetra.freeze/csharp/ICT/Petra/Client/MReporting/logic/Calculator.cs	2014-07-24 14:18:45.000000000 +0200
+++ openpetra.patched/csharp/ICT/Petra/Client/MReporting/logic/Calculator.cs	2014-07-24 14:21:43.000000000 +0200
@@ -61,7 +61,7 @@
 
         /// <summary>how long did it take to calculate the report</summary>
         protected TimeSpan Duration;
-        private IReportingUIConnectorsReportGenerator FReportingGenerator;
+        private TMReportingNamespace.TReportingUIConnectorsNamespace.TReportGeneratorUIConnector FReportingGenerator;
         private Boolean FKeepUpProgressCheck;
 
         /// <summary>
@@ -469,11 +469,9 @@
             Thread ProgressCheckThread;
 
             ReturnValue = false;
-            FReportingGenerator = TRemote.MReporting.UIConnectors.ReportGenerator();
+            FReportingGenerator = (TMReportingNamespace.TReportingUIConnectorsNamespace.TReportGeneratorUIConnector)TRemote.MReporting.UIConnectors.ReportGenerator();
             FKeepUpProgressCheck = true;
 
-            // Register Object with the TEnsureKeepAlive Class so that it doesn't get GC'd
-            TEnsureKeepAlive.Register(FReportingGenerator);
             try
             {
                 this.Results = new TResultList();
@@ -485,8 +483,8 @@
             {
                 TLogging.Log(e.Message);
 
-                // UnRegister Object from the TEnsureKeepAlive Class so that the Object can get GC'd on the PetraServer
-                TEnsureKeepAlive.UnRegister(FReportingGenerator);
+                // 'Release' instantiated UIConnector Object on the server side so it can get Garbage Collected there
+                TUIConnectorLifetimeHandling.ReleaseUIConnector(FReportingGenerator);
             }
 
             // todo: allow canceling of the calculation of a report
@@ -522,9 +520,6 @@
 
         private void AsyncProgressCheckThread()
         {
-            TAsyncExecProgressState ProgressState;
-            Int16 ProgressPercentage;
-            String ProgressInformation;
             String OldLoggingText;
             DateTime startTime;
 
@@ -533,38 +528,34 @@
 
             while (FKeepUpProgressCheck)
             {
-                FReportingGenerator.AsyncExecProgress.ProgressCombinedInfo(out ProgressState, out ProgressPercentage, out ProgressInformation);
+                TProgressState state = FReportingGenerator.Progress;
 
-                switch (ProgressState)
+                if (state.JobFinished)
                 {
-                    case TAsyncExecProgressState.Aeps_Finished:
-                        this.Duration = DateTime.Now - startTime;
+                    this.Duration = DateTime.Now - startTime;
 
-                        if (FReportingGenerator.GetSuccess() == true)
-                        {
-                            this.Parameters.LoadFromDataTable(FReportingGenerator.GetParameter());
-                            this.Results.LoadFromDataTable(FReportingGenerator.GetResult());
-                            this.Results.SetMaxDisplayColumns(this.Parameters.Get("MaxDisplayColumns").ToInt());
-                        }
-                        else
-                        {
-                            TLogging.Log(FReportingGenerator.GetErrorMessage());
-                        }
-
-                        // UnRegister Object from the TEnsureKeepAlive Class so that the Object can get GC'd on the PetraServer
-                        TEnsureKeepAlive.UnRegister(FReportingGenerator);
-                        FKeepUpProgressCheck = false;
-                        break;
-
-                    default:
-
-                        if ((ProgressInformation != null) && (!OldLoggingText.Equals(ProgressInformation)))
-                        {
-                            TLogging.Log(ProgressInformation, TLoggingType.ToStatusBar);
-                            OldLoggingText = ProgressInformation;
-                        }
+                    if (FReportingGenerator.GetSuccess() == true)
+                    {
+                        this.Parameters.LoadFromDataTable(FReportingGenerator.GetParameter());
+                        this.Results.LoadFromDataTable(FReportingGenerator.GetResult());
+                        this.Results.SetMaxDisplayColumns(this.Parameters.Get("MaxDisplayColumns").ToInt());
+                    }
+                    else
+                    {
+                        TLogging.Log(FReportingGenerator.GetErrorMessage());
+                    }
 
-                        break;
+                    // 'Release' instantiated UIConnector Object on the server side so it can get Garbage Collected there
+                    TUIConnectorLifetimeHandling.ReleaseUIConnector(FReportingGenerator);
+                    FKeepUpProgressCheck = false;
+                }
+                else
+                {
+                    if ((state.StatusMessage != null) && (!OldLoggingText.Equals(state.StatusMessage)))
+                    {
+                        TLogging.Log(state.StatusMessage, TLoggingType.ToStatusBar);
+                        OldLoggingText = state.StatusMessage;
+                    }
                 }
 
                 if (FKeepUpProgressCheck)
