--- openpetraNow.freeze2014_05/setup/petra0300/remoteclientWin/remote.nsi	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/setup/petra0300/remoteclientWin/remote.nsi	2014-05-15 22:04:07.000000000 +0200
@@ -98,7 +98,6 @@
   File ..\..\..\csharp\ThirdParty\A1Panel\Owf.Controls.A1Panel.dll
   File ..\..\..\csharp\ThirdParty\OrientedTextControls\CustomControl.OrientedTextControls.dll
   File ..\..\..\delivery\bin\Ict.Common*dll
-  File ..\..\..\delivery\bin\Ict.Petra.Server.*.connect.dll
   File ..\..\..\delivery\bin\Ict.Petra.Client*dll
   File ..\..\..\delivery\bin\Ict.Petra.Shared*dll
   File ..\..\..\delivery\bin\PetraClient.exe
@@ -135,9 +134,7 @@
   CreateShortCut "$SMPROGRAMS\${MUI_PRODUCT}\OpenPetra.org Client.lnk" "$INSTDIR\bin30\PetraClient.exe" '-C:"$INSTDIR\etc30\PetraClientRemote.config"' $INSTDIR\petraico-big.ico 0 SW_SHOWNORMAL
   ; avoid problems with empty hotkey. so no comment for the moment for the shortcut: "Start OpenPetra.org (connecting to your OpenPetra server)"
   CreateShortCut "$SMPROGRAMS\${MUI_PRODUCT}\Uninstall.lnk" "$INSTDIR\Uninstall.exe"
-  
-  ExecWait '"cmd.exe" /C netsh firewall set allowedprogram program = "$INSTDIR\bin30\PetraClient.exe" name = PetraClient mode = DISABLE'
-  
+
   ;Create uninstaller
   WriteUninstaller "$INSTDIR\Uninstall.exe"
 SectionEnd
