--- openpetraNow.freeze2014_05/setup/petra0300/remoteclientWin/remote.iss	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/setup/petra0300/remoteclientWin/remote.iss	2014-05-15 22:04:07.000000000 +0200
@@ -174,9 +174,4 @@
     ReplaceInTextFile(ExpandConstant('{userappdata}/OpenPetra{#ORGNAME}/etc30/PetraClientRemote.config'), 'PETRAHOST', strServer, true);
     ReplaceInTextFile(ExpandConstant('{userappdata}/OpenPetra{#ORGNAME}/etc30/PetraClientRemote.config'), 'PETRAPORT', IntToStr(NetPort), true);
   end;
-
-  // allow the .net remoting communication between client and server
-  Exec(ExpandConstant('{sys}\cmd.exe'), '/C netsh firewall set allowedprogram program = '
-    + ExpandConstant('"{userappdata}/OpenPetra{#ORGNAME}\bin30\PetraClient.exe" name = PetraClient mode = DISABLE'),
-    '', SW_HIDE, ewWaitUntilTerminated, ResultCode);
 end;
