--- openpetraNow.freeze2014_05/csharp/ICT/Testing/lib/MPartner/server/PartnerEdit/TestPartnerEdit.cs	2014-06-23 12:34:11.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Testing/lib/MPartner/server/PartnerEdit/TestPartnerEdit.cs	2014-06-28 06:58:29.000000000 +0200
@@ -63,20 +63,13 @@
     public class TPartnerEditTest
     {
         /// <summary>
-        /// use automatic property to avoid compiler warning about unused variable FServerManager
-        /// </summary>
-        private TServerManager FServerManager {
-            get; set;
-        }
-
-        /// <summary>
         /// open database connection or prepare other things for this test
         /// </summary>
         [TestFixtureSetUp]
         public void Init()
         {
             new TLogging("../../log/TestServer.log");
-            FServerManager = TPetraServerConnector.Connect("../../etc/TestServer.config");
+            TPetraServerConnector.Connect("../../etc/TestServer.config");
         }
 
         /// <summary>
