--- openpetraNow.freeze2014_05/inc/nanttasks/NUnitConsole.cs	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/inc/nanttasks/NUnitConsole.cs	2014-05-15 22:04:07.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -99,7 +99,7 @@
 
                 process.StartInfo.FileName = exeName;
 
-                process.StartInfo.Arguments = "\"" + FAssemblyName + "\"";
+                process.StartInfo.Arguments = "\"" + FAssemblyName + "\" /result=../../log/TestResult.xml";
             }
 
             if (FTestCase.Length > 0)
