--- openpetraNow.freeze2014_05/csharp/ICT/Common/FileVersion.cs	2014-05-16 08:51:14.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Common/FileVersion.cs	2014-05-15 22:04:07.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -102,8 +102,8 @@
         {
             FileMajorPart = System.Convert.ToUInt16(AVersion.Major);
             FileMinorPart = System.Convert.ToUInt16(AVersion.Minor);
-            FileBuildPart = System.Convert.ToUInt16(AVersion.Build);
-            FilePrivatePart = System.Convert.ToUInt16(AVersion.Revision);
+            FileBuildPart = System.Convert.ToUInt16(AVersion.Build == -1 ? 0 : AVersion.Build);
+            FilePrivatePart = System.Convert.ToUInt16(AVersion.Revision == -1 ? 0 : AVersion.Revision);
         }
 
         /// <summary>
@@ -126,6 +126,15 @@
         }
 
         /// <summary>
+        /// convert to System.Version
+        /// </summary>
+        /// <returns></returns>
+        public Version ToVersion()
+        {
+            return new Version(FileMajorPart, FileMinorPart, FileBuildPart, FilePrivatePart);
+        }
+
+        /// <summary>
         /// compare two file versions
         /// </summary>
         /// <returns>-1 if this &lt; ACmp, 1 if this &gt; ACmp, and 0 if equals</returns>
@@ -232,6 +241,11 @@
         }
 
         /// <summary>
+        /// useful for testing, see Login.test.cs
+        /// </summary>
+        public static TFileVersionInfo FManualApplicationVersion = null;
+
+        /// <summary>
         /// get the version of the current application.
         /// Parse version.txt in the same directory if that file exists.
         /// Otherwise use the version of the exe or dll file
@@ -241,6 +255,11 @@
         {
             TFileVersionInfo Result = new TFileVersionInfo();
 
+            if (FManualApplicationVersion != null)
+            {
+                return FManualApplicationVersion;
+            }
+
             // retrieve the current version of the server from the file version.txt in the bin directory
             // this is easier to manage than to check the assembly version in case you only need to quickly update the client
             string BinPath = TAppSettingsManager.ApplicationDirectory;
