--- openpetraNow.freeze2014_05/csharp/ICT/Petra/Server/lib/MPartner/Common/Extracts.cs	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Petra/Server/lib/MPartner/Common/Extracts.cs	2014-05-15 22:04:07.000000000 +0200
@@ -44,9 +44,7 @@
     /// These Business Objects handle the retrieval, verification and saving of data.
     ///
     /// @Comment These Business Objects can be instantiated by other Server Objects
-    ///          (usually UIConnectors) and also directly from the Instantiator
-    ///          classes (only when it makes sense AND they derive from
-    ///          TConfigurableMBRObject).
+    ///          (usually UIConnectors).
     /// </summary>
     public static class TExtractsHandling
     {
