--- openpetraNow.freeze2014_05/csharp/ICT/Petra/Server/lib/MCommon/IntranetExport.cs	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Petra/Server/lib/MCommon/IntranetExport.cs	2014-05-15 22:04:07.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       Tim Ingham
 //
-// Copyright 2012 by OM International
+// Copyright 2013 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -480,7 +480,7 @@
 
         private static Boolean ExportField()
         {
-            Int64 MySiteKey = DomainManagerBase.GSiteKey;
+            Int64 MySiteKey = DomainManager.GSiteKey;
 
             /*
              *  From 4GL:
@@ -713,7 +713,7 @@
 
             sw.WriteLine("key,value");
             sw.WriteLine("version," + ExportVersion);
-            sw.WriteLine("office," + ((Int64)DomainManagerBase.GSiteKey).ToString("D10"));
+            sw.WriteLine("office," + ((Int64)DomainManager.GSiteKey).ToString("D10"));
             sw.WriteLine("date," + DateTime.Now.ToString(ExportDateFormat));
             sw.WriteLine("time," + DateTime.Now.ToString("HH:mm:ss"));
             sw.WriteLine("options," + AOptionalMetadata);
@@ -817,7 +817,7 @@
             {
                 FZipFileNames.Clear();
                 FExportTrace = "Exporting\r\n";
-                FExportFilePath = TAppSettingsManager.GetValue("OpenPetra.PathTemp") + @"\";
+                FExportFilePath = TAppSettingsManager.GetValue("Server.PathTemp") + @"\";
                 FTransaction = DBAccess.GDBAccessObj.BeginTransaction(IsolationLevel.ReadCommitted);
 
                 if (AExportDonationData)
