--- openpetra.freeze/csharp/ICT/Petra/Server/lib/MFinance/connect/Gift.GiftDetailFind.cs	2014-07-24 14:18:45.000000000 +0200
+++ openpetra.patched/csharp/ICT/Petra/Server/lib/MFinance/connect/Gift.GiftDetailFind.cs	2014-07-30 20:08:07.000000000 +0200
@@ -4,7 +4,7 @@
 // @Authors:
 //       christiank, timop
 //
-// Copyright 2004-2012 by OM International
+// Copyright 2004-2014 by OM International
 //
 // This file is part of OpenPetra.org.
 //
@@ -55,7 +55,7 @@
     ///          However, Server Objects that derive from these objects and that
     ///          are also UIConnectors are feasible.
     /// </summary>
-    public class TGiftDetailFindUIConnector : TConfigurableMBRObject, IFinanceUIConnectorsGiftDetailFind
+    public class TGiftDetailFindUIConnector : IFinanceUIConnectorsGiftDetailFind
     {
         private TGiftDetailFind FGiftDetailFind = new TGiftDetailFind();
 
@@ -66,14 +66,12 @@
         {
         }
 
-        /// <summary>Returns reference to the Asynchronous execution control object to the caller</summary>
-        public IAsynchronousExecutionProgress AsyncExecProgress
+        /// <summary>Get the current state of progress</summary>
+        public TProgressState Progress
         {
             get
             {
-                return (IAsynchronousExecutionProgress)TCreateRemotableObject.CreateRemotableObject(
-                    typeof(TAsynchronousExecutionProgressRemote),
-                    FGiftDetailFind.AsyncExecProgress);
+		return FGiftDetailFind.Progress;
             }
         }
 
@@ -130,4 +128,4 @@
             FGiftDetailFind.StopSearch(ASender, AArgs);
         }
     }
-}
\ No newline at end of file
+}
