--- openpetraNow.freeze2014_05/csharp/ICT/Petra/ServerPlugins/AuthenticationPhpBB/authenticate.php	2014-05-16 08:51:15.000000000 +0200
+++ openpetraNow.PatchRemoting/csharp/ICT/Petra/ServerPlugins/AuthenticationPhpBB/authenticate.php	2014-05-15 22:04:07.000000000 +0200
@@ -43,8 +43,8 @@
 $mycr = new mycrypt();
 // make sure the path of secretkey.dat cannot be accessed from the outside
 $mycr->init($SecretKeyFile);
-$msg = base64_decode($_GET["msg"]);
-$iv = base64_decode($_GET["msg2"]);
+$msg = base64_decode($_POST["msg"]);
+$iv = base64_decode($_POST["msg2"]);
 $clearmsg = $mycr->decrypt($msg, $iv);
 $pos = strpos($clearmsg, ";");
 $username = substr($clearmsg, 0, $pos);
